### PR TITLE
Refactor ColorWebSDK api interface and request fixes

### PR DESCRIPTION
**Pull request by scwe**

### CODE REVIEW

File: src/config/package.json [hunk 1]
```diff
@@ -0,0 +1 @@
+./../../package.json
```
Comment (scwe) [inline src/config/package.json]: This symlink is perhaps not the most useful... Haha

File: src/index.js [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+  element,
+  callbacks
 }
+const sdkInstance1 = new ColorWebSDK(sdkParams)
 
-video.onpause = () => {
-  firstInstance.showRecommendations()
+const showRecommendations1 = () => {
+  sdkInstance1
+    .getRecommendations()
```
Comment (scwe) [inline src/index.js]: I am still not convinced we need to force the user to ``getRecommendations``

File: src/utils/session/index.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// LIBS
+import * as Cookies from 'js-cookie'
+import debug from 'debug'
+// LOGIC
+import initSessions from './init'
+import keepSession from './keep-session'
+import executeFunction from './../execute-if-is-function'
+// CONST
+const COOKIES_KEY = 'cWebSDK-session'
+const MILIECONDS = 1000
```
Comment (scwe) [inline src/utils/session/index.js]: Yeah... That is not how you spell ``MILLISECONDS``

File: src/color-web-sdk.js [hunk 4]
```diff
@@ -7,80 +7,86 @@ import './color-web-sdk.css'
...
+import ColorWebSDKSession from './utils/session'
 import getRecommendations from './utils/recommendations/get-recommendations'
-
+import bringIntoView from './utils/scroll/scroll-to-be-visible'
+import executeFunction from './utils/execute-if-is-function'
 // CONST
-const MILIECONDS = 1000
+import { scroll } from './config'
+const SCROLL_ANIMATION = scroll.durationMs
+const SESSIONS = {}
```
Comment (scwe) [inline src/color-web-sdk.js]: I'm inclined to think that ``SESSIONS`` should be stored in the file dedicated to sessions

File: src/index.js [hunk 5]
```diff
@@ -1,16 +1,39 @@
...
+const video = document.querySelector('.container video')
+const appId = ''
+const partnerVideoId = ''
+const callbacks = {
+  onRecommendationsLoaded: () => logR('on load'),
+  onRecommendationsShow: () => logR('on show'),
+  onRecommendationsClose: () => logR('on close'),
+  onSessionInit: () => logS('on initialized'),
+  onSessionUpdate: () => logS('on update'),
+  onScrollToFinished: () => logST('on finished')
```
Comment (scwe) [inline src/index.js]: I always kinda preferred lists of callbacks that you can add and remove from. But the user can implement that if they want it

File: src/config/package.json (hunk 1 above)
Comment (psmarcin) [inline src/config/package.json]: ha ha ha didnt noticed that :D it is how git works :P

File: src/index.js (hunk 5 above)
Comment (psmarcin) [inline src/index.js]: `callbacks` field is options, you also can use only some part of callbacks

File: src/utils/session/index.js (hunk 3 above)
Comment (psmarcin) [inline src/utils/session/index.js]: ohhh, sorry :( will fix in a minute

File: src/color-web-sdk.js (hunk 4 above)
Comment (psmarcin) [inline src/color-web-sdk.js]: I put it here for performance reason. If I will store it in Session file we will have to create instance of `ColorWebSDKSession` for each video, right now we only create one per appId session

**Comment by psmarcin** [comment]
@scwe am I answer to all your questions?

File: src/index.js (hunk 2 above)
Comment (psmarcin) [inline src/index.js]: When we will get feedback from CBS maybe we will change it to your idea.

File: src/color-web-sdk.js (hunk 4 above)
Comment (scwe) [inline src/color-web-sdk.js]: Not really, it can still be a constant in the file right?

File: src/index.js (hunk 5 above)
Comment (scwe) [inline src/index.js]: No like, onSessionInit is a list of callbacks rather than just one. But that can wait till another time

**Comment by wojciechAdaszynski** [comment]
@psmarcin You need to add polyfill to support older browsers

**Comment by psmarcin** [comment]
@wojciechAdaszynski done

**Pull request merged**